CMakeLists.txt

```diff
@@ -11,6 +11,8 @@ set(TARGET_LIBRARY beman_${TARGET_NAME})
 set(TARGET_ALIAS beman::${TARGET_NAME})
 set(TARGETS_EXPORT_NAME ${CMAKE_PROJECT_NAME})
 
+option(BEMAN_NET_WITH_URING "Enable liburing io context" OFF)
```

Comment:
On Linux there are at least 4 sequencers: `select`, `poll`, `epoll_*`, and `iouring_*`. On other systems there are additional sequence, e.g. `kqueue*` and IOCP. The interface `context_base` and `io_context` are set up to make it an object creation choice which sequencer is used. My vision on setting this up is to have `cmake`-level checks to determine what implementations are being _build_.  There should be a default sequencer and that may either be prioritized based on what sequencers can be build, an `cmake`-option, or a combination thereof.

I can provide examples of how I think that could look like.